mysite/announcements/tasks.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import absolute_import, unicode_literals
+
+from celery import task
+from django.utils import timezone
+
+from .models import Announcements
+
+
+@task()
+def check_scheduled_announcements():
+    current_time = timezone.now()
+    scheduled_announcement = Announcements.objects.filter(date_time_to_publish__lte=current_time)\
+        .filter(sent_at=None)\
+        .prefetch_related('groups__user_set')
+
+    announcement_id = []
+    for announcement in scheduled_announcement:
+        group_list = announcement.groups.all()
+        user_list = []
+
+        for group in group_list:
+            users = group.user_set.all()
+            for u in users:
+                user_list.append(u)
```

Comment:
We can use `list comprehension` here.

It is proven that list comprehension is faster than calling `.append` in the loop.

http://www.pythonforbeginners.com/basics/list-comprehensions-in-python
https://www.programiz.com/python-programming/list-comprehension


Above can be transformed into 
```py
users_list = [user for group in groups for user in group]
```